static/app/components/organizations/globalSelectionHeader/index.tsx

```diff
@@ -53,41 +71,92 @@ function GlobalSelectionHeaderContainer({
 
   const nonMemberProjects = isSuperuser || isOrgAdmin ? otherProjects : [];
 
-  const enforceSingleProject = !organization.features.includes('global-views');
+  // Initializes GlobalSelectionHeader
+  //
+  // Calls an actionCreator to load project/environment from local storage if possible,
+  // otherwise populate with defaults.
+  //
+  // This should only happen when the header is mounted
+  // e.g. when changing views or organizations.
+  useEffect(() => {
```

Comment:
if we really wanna get fancy we can move this into it's own hook

Reply:
That seems elaborate given we have no other use cases for the custom hook.

Reply:
yeah agreed there.

Was suggesting this more as a way to encapsulate this logic in a different file